contract/vm.go

```diff
@@ -825,57 +829,71 @@ func Call(
 	if err != nil {
 		return "", nil, ctx.usedFee(), err
 	}
+
 	if ctrLgr.IsDebugEnabled() {
-		ctrLgr.Debug().Str("abi", string(code)).Str("contract", types.EncodeAddress(contractAddress)).Msg("call")
+		ctrLgr.Debug().Str("abi", string(payload)).Str("contract", types.EncodeAddress(contractAddress)).Msg("call")
 	}
 
+	// create a new executor
 	contexts[ctx.service] = ctx
 	ce := newExecutor(contract, contractAddress, ctx, &ci, ctx.curContract.amount, false, false, contractState)
 	defer ce.close()
 
+	// execute the contract call
 	ce.call(callMaxInstLimit, nil)
+
+	// check if there is an error
 	err = ce.err
 	if err != nil {
+		// rollback the state of the contract
 		if dbErr := ce.rollbackToSavepoint(); dbErr != nil {
 			ctrLgr.Error().Err(dbErr).Str("contract", types.EncodeAddress(contractAddress)).Msg("rollback state")
 		}
+		// log the error
 		if ctx.traceFile != nil {
 			_, _ = ctx.traceFile.WriteString(fmt.Sprintf("[error] : %s\n", err))
 			_, _ = ctx.traceFile.WriteString(fmt.Sprintf("[usedFee] : %s\n", ctx.usedFee().String()))
-			evs := ce.getEvents()
-			if evs != nil {
+			events := ce.getEvents()
+			if events != nil {
 				_, _ = ctx.traceFile.WriteString("[Event]\n")
-				for _, ev := range evs {
-					eb, _ := ev.MarshalJSON()
-					_, _ = ctx.traceFile.Write(eb)
+				for _, event := range events {
+					event_json, _ := event.MarshalJSON()
```

Comment:
golang generally uses camelCase. ( some linter warn snake_case. ) I think jsonEvent is better (but event_json is no problem too)

Reply:
OK